File: swiftcookui/src/interfaces/recipeingredient.ts
import type { UnitDto } from './unit'

export interface RecipeIngredientDto {
  recipeId: number
  ingredientId: number
  ingredientName: string
  amount: number | null
  unit?: UnitDto
  position: number
}

export interface RecipeIngredientCreateDto {
  ingredientId: number
  amount: number | null
  unitId: number | null
  position: number
}

File: swiftcookdb/Models/RecipeIngredient.cs
using SwiftCookDb.Models;

namespace SwiftCookDb.Models
{
    public class RecipeIngredient
    {
        public int RecipeId { get; set; }
        public Recipe Recipe { get; set; } = null!;

        public int IngredientId { get; set; }
        public Ingredient Ingredient { get; set; } = null!;

        public int? UnitId { get; set; }
        public Unit Unit { get; set; } = null!;

        public float? Amount { get; set; }
        public int? Position { get; set; }
    }
}